FiftyOne integration (WIP)

benjaminpkane opened the pull request:
Updates required by and suitable for integrating player51 into [FiftyOne](https://github.com/voxel51/fiftyone).

Player51 is intended to be the core media visualization tool for FiftyOne.

benjaminpkane committed:
better positioning

benjaminpkane committed:
stripping away sizing logic

lethosor commented on src/js/renderers/imagerenderer.js:
```diff
@@ -137,7 +137,7 @@ ImageRenderer.prototype.initPlayerControls = function() {
   });
 
-  this.updateFromLoadingState();
+  // this.updateFromLoadingState();
```
Intentional?

benjaminpkane replied:
Yes this is the crux of my issue from yesterday. The images test in p51 loads the overlay asynchronously. FiftyOne already has the overlaydata when initializing a player.

`ObjectOverlay.setup` was being called with an undefined `canvasWidth` and `canvasHeight`.

lethosor replied:
Hm, I think the general pattern here is that the `updateFrom*` functions are responsible for double-checking the state, not the callers. I'm seeing `updateFromLoadingState` called in other places too, though, so as long as this doesn't break loading behavior for the standalone examples, it should be fine either way.

benjaminpkane committed:
tweaks

benjaminpkane committed:
getting rid of build errors

benjaminpkane committed:
fix error

benjaminpkane committed:
inserting custom color map

benjaminpkane committed:
bbox toggling work

benjaminpkane committed:
toggle functionality

benjaminpkane committed:
testing

benjaminpkane committed:
prevent image smoothing

lethosor commented on src/css/player51.css:
```diff
@@ -297,6 +301,36 @@
+  width: 100%;
+  height: 100%;
+
+  image-rendering: optimizeSpeed;             /* STOP SMOOTHING, GIVE ME SPEED  */
+  image-rendering: -moz-crisp-edges;          /* Firefox                        */
+  image-rendering: -o-crisp-edges;            /* Opera                          */
+  image-rendering: -webkit-optimize-contrast; /* Chrome (and eventually Safari) */
+  image-rendering: pixelated; /* Chrome */
+  image-rendering: optimize-contrast;         /* CSS3 Proposed                  */
+  -ms-interpolation-mode: nearest-neighbor;   /* IE8+                           */
```
I'm not sure that we'll want this for every image (e.g. on scoop, etc) - maybe this could be configurable?

benjaminpkane replied:
Sure, I can work that in.

benjaminpkane committed:
load lazily

benjaminpkane committed:
rm unused colorMap

Commit:
Revert "rm unused colorMap"

Breaks colors of detections

This reverts commit cf5e076b20c8bcc8692cee6d9ef09aecacdc9d23.

Commit:
FrameAttributesOverlay: add support for activeLabels

Commit:
Hide object attribute options if no objects have attributes

Commit:
Unconditionally call _parseAttrs (previous check was a no-op)

Commit:
Allow clicking on objects without IDs

Commit:
Upgrade dependencies

Commit:
Ignore .css.d.ts files (generated by typed-css-modules-webpack-plugin)

benjaminpkane committed:
filter work

benjaminpkane committed:
filter checks for overlays

benjaminpkane committed:
add confidence to object overlay

Commit:
Hide frame attribute overlay if no attributes are visible

Closes voxel51/fiftyone#421

benjaminpkane committed:
update filter check

benjaminpkane committed:
fix logic

Commit:
Improve robustness a bit

benjaminpkane committed:
use grid filters, for now

benjaminpkane committed:
tweaks for filter

Commit:
Fix syntax error

benjaminpkane committed:
fix syntax

benjaminpkane committed:
Merge branch 'fo-work' of github.com:voxel51/player51 into fo-work

benjaminpkane committed:
use name

Commit:
Pass colorMap and fps through options object

Commit:
Move activeLabels and filter to options, add updateOptions()

Commit:
Add option to show object confidence

Commit:
Add confidence to frame attributes

Commit:
Fix filtering of objects with no confidence

Commit:
Make clicking within player hide options menu

Commit:
Change default overlay options to only show hovered object

Commit:
Disable object clicking/hovering in thumbnail mode

Commit:
Lint

Commit:
Prevent hidden objects from handling click/hover events

benjaminpkane committed:
ObjectOverlay: always use renderer colorMap

Commit:
Fix visibility of attribute options when attributes are present

Commit:
Allow overriding default overlay options

Commit:
Allow disabling overlay options

Commit:
Restore default overlay options

Reverts 274591bd9c6a21a490988757753a32606858a63a

Commit:
Store `action` as string, not object

Commit:
Update label when hover is default

Commit:
Remove "display options" header

Commit:
Add support for getting/setting overlay options

Commit:
Fix VideoRenderer options initialization

Commit:
Expose content dimensions to clients

Commit:
Fire custom load event when media is loaded

Closes #41

benjaminpkane committed:
color for video labels

Commit:
Support empty overlays passed as objects

These only previously worked when passed as a URL - an empty object, {}, would be ignored, resulting in an unplayable video

benjaminpkane committed:
override isShown for video to check frames key

benjaminpkane committed:
Merge branch 'fo-work' of github.com:voxel51/player51 into fo-work

Commit:
Send error event when video fails to load

Commit:
Fix mask coloring by calculating raw color on the fly

Commit:
Cache mask color computations

Commit:
Add support for multiple frame-level masks

Commit:
Add filtering support for frame-level masks

Commit:
Make mask smoothing configurable

Commit:
Add support for rendering keypoints

Commit:
Pass through keypoints in video labels

Commit:
Refactor some drawing constants

benjaminpkane committed:
always define frame number, and render first overlay in initial paused thumbnail

benjaminpkane committed:
lint

benjaminpkane committed:
process frame at the end of an update

Commit:
Add support for rendering polylines

Commit:
Ensure that frame attributes render above other overlays (specifically filled polylines)

Commit:
Use transparent fill for polylines

Commit:
Fix visibility of keypoints/polylines in video frames

Commit:
Fix colors of keypoints/polylines in video frames

benjaminpkane committed:
simplify _prepareOverlay_auxCheckAdd

benjaminpkane committed:
callback for mouseenter event

Commit:
Lint

Commit:
Implement containsPoint() for polylines

benjaminpkane committed:
update onMouseEnter placement

benjaminpkane committed:
Merge branch 'fo-work' of github.com:voxel51/player51 into fo-work

Commit:
Add support for sending events when an object is selected

benjaminpkane committed:
Merge branch 'obj-selection' into fo-work

benjaminpkane committed:
dispatch event

Commit:
Refactor polyline overlay to use one instance per object

benjaminpkane committed:
hovering state

Commit:
Refactor keypoint overlay to use one instance per object

benjaminpkane committed:
remove frames special case

benjaminpkane committed:
formalizing hover loading with public methods

benjaminpkane committed:
linting

benjaminpkane committed:
absolute positioning for div.p51-contained-image

benjaminpkane committed:
deep copy new overlays

Commit:
Merge branch 'obj-selection' into fo-work

Commit:
Lint

Commit:
Also clear frameOverlay in updateOverlay() to fix duplicate objects in videos

Commit:
Add support for multi-shape polylines

Commit:
Merge branch 'multi-polylines' into fo-work

benjaminpkane committed:
update fps settings if necessary

Commit:
Merge branch 'fo-work' into obj-selection

Commit:
Draw selected objects with dashed border

Commit:
Add support for selecting keypoints

Commit:
Account for object visibility when selecting keypoints/polylines

Commit:
Clear focus more reliably when mouse leaves canvas

Commit:
Implement selection for non-closed and non-filled polylines

Commit:
Propagate timeupdate event

Commit:
Include overlay name in "select" event payload

Commit:
Add support for prioritizing mouse events over object borders

Commit:
ObjectOverlay: Implement border detection

Commit:
Support border detection for closed/filled polylines

Commit:
Make object header detection a bit more tolerant

Commit:
Merge branch 'obj-contains-mode' into obj-selection

benjaminpkane committed:
Merge branch 'obj-selection' into fo-work

benjaminpkane committed:
Merge branch 'obj-selection' into fo-work

Commit:
Merge branch 'obj-selection' into fo-work

benjaminpkane committed:
Merge branch 'fo-work' of github.com:voxel51/player51 into fo-work

benjaminpkane committed:
coloredByLabel option

benjaminpkane committed:
color by label logic

benjaminpkane committed:
change to global color by label

benjaminpkane committed:
linting

benjaminpkane committed:
check for renderer before removing removing event listeners

benjaminpkane committed:
lint

benjaminpkane committed:
expose getColor

benjaminpkane committed:
remove code comment

benjaminpkane committed:
fps allback

benjaminpkane committed:
remove updateFromLoadingState call in videorenderer

benjaminpkane committed:
update parent styles in some tests

benjaminpkane committed:
move dependencies to devDependencies

benjaminpkane committed:
ungap fevent target

benjaminpkane committed:
Merge branch 'develop' into fo-work

benjaminpkane committed:
update control checkboxes on overlay options update

benjaminpkane committed:
prototyping point info api

benjaminpkane committed:
point info work

benjaminpkane committed:
point info tweaks

benjaminpkane committed:
filling out pointInfo API

benjaminpkane committed:
adding frame mask utility functions

benjaminpkane committed:
classification support

benjaminpkane committed:
tweaks

benjaminpkane committed:
targets in overlays, keypoint improvements, disable thumbnail tooltips

benjaminpkane committed:
fix controls bar interefence

benjaminpkane committed:
mouse distance work

benjaminpkane committed:
more distance work

benjaminpkane committed:
finalizing tooltip events

benjaminpkane committed:
zero distance if mouse is within keypoint radius

benjaminpkane committed:
support null labels in object overlays

benjaminpkane committed:
rm None text in object overlay

benjaminpkane committed:
dispatch error events from img and video elements

benjaminpkane committed:
merge null-labels

benjaminpkane committed:
prevent arrow and scroll z order events from propagating

benjaminpkane committed:
find hovered overlays

benjaminpkane committed:
options change event

benjaminpkane committed:
merge null-labels

benjaminpkane committed:
only rotate hovered objects

benjaminpkane committed:
bug fixes, no rotate events for thumbnails

benjaminpkane committed:
always compute min mouse distance for keypoints

benjaminpkane committed:
rotation fixes, cleanup

benjaminpkane committed:
focus tweaks

benjaminpkane committed:
order persistence, top object work

benjaminpkane committed:
update overlay order before event handling

benjaminpkane committed:
bug fixes

benjaminpkane committed:
best overlay fixes

benjaminpkane committed:
rotation fixes

benjaminpkane committed:
update active labels usage

benjaminpkane committed:
frame attrs work

benjaminpkane committed:
font work

benjaminpkane committed:
font tweaks, bug fixes

benjaminpkane committed:
isShown fix

benjaminpkane committed:
activeLabels overlay ordering

benjaminpkane committed:
integrating ordered overlays

benjaminpkane committed:
ordering flow adjustments

benjaminpkane committed:
bug fixes

benjaminpkane committed:
bug fixes

benjaminpkane committed:
fixing/simplifying hover ordering

benjaminpkane committed:
perfection!

benjaminpkane committed:
handle empty case, segmentations fix

benjaminpkane committed:
more bug fixes

benjaminpkane committed:
showTooltip option

benjaminpkane committed:
bug fix

benjaminpkane committed:
revert font changes

benjaminpkane committed:
fix colorByLabel

benjaminpkane committed:
wall rotations, remove wheel input

benjaminpkane committed:
wall fix

benjaminpkane committed:
include tooltip in updateOverlayOptions

benjaminpkane committed:
include _id and name when preparing mask overlay

benjaminpkane committed:
full opacity for selected masks

benjaminpkane committed:
getSelectData

benjaminpkane committed:
classification blocks

benjaminpkane committed:
text box fixes

benjaminpkane committed:
attr selection work

benjaminpkane committed:
get single attr point info

benjaminpkane committed:
bug fix

benjaminpkane committed:
bug fixes, selection events for classifications

benjaminpkane committed:
classification selection border

benjaminpkane committed:
classification selection border fixes

benjaminpkane committed:
cleanup frame selection fix

benjaminpkane committed:
smaller border for classifications

benjaminpkane committed:
fix selection and overlay ordering interference

benjaminpkane committed:
colorGenerator option

benjaminpkane committed:
segmentation rendering fixes

benjaminpkane committed:
return field name as name frame attr select data

benjaminpkane committed:
hover load fix

benjaminpkane committed:
frame number in point info

benjaminpkane committed:
mouseenter fix

benjaminpkane committed:
use seed for colorGenerator

benjaminpkane committed:
store tags in overlays

benjaminpkane committed:
overlays refresh

benjaminpkane committed:
cleaning

benjaminpkane committed:
image overlay bug fixes

benjaminpkane committed:
simplifying api

benjaminpkane committed:
cleaning

benjaminpkane committed:
refactor with composition for readability

benjaminpkane committed:
refactor
